src/slash_env_var_cmds.c

```diff
@@ -0,0 +1,190 @@
+#include <slash/slash.h>
+#include <slash/optparse.h>
+#include <stdlib.h>
+#include <string.h>
+#include "environment.h"
+
+/**
+ * Adds the following commands to CSH:
+ * - var_set
+ * - var_get
+ * - var_unset
+ * - var_show
+ * - var_clear
+ * - var_expand
+ * 
+ * Examples:
+ * 
+ * var_set WATCH_TIME 2000
+ * var_set CMD "list"
+ * var_show
+ *
+ * var_expand "watch -n $(WATCH_TIME) $(CMD)"
+ * - prints the expanded version of its argument
+ * 
+ * var_expand -e "watch -n $(WATCH_TIME) $(CMD)"
+ * -e means "csh-execute the expanded argument"
+ * 
+ * var_unset CMD
+ * var_show
+ * 
+ * var_clear
+ * var_show
+ * 
+ * var_set MY_PASSWORD 123456 # Don't you ever do that!
+ * var_expand -q -e "command -p $(PASSWORD)"
+ * -q means do NOT print the expanded version of its argument, but just csh-execute it.
+ * This is useful of you do not want the "PASSWORD" variable to appear in logs, for instance. The log will still
+ * contain this 'var_expand -q -e "command -p $(PASSWORD)"', which is still useful enough for tracing/debugging without
+ * leaking sensitive info.
+ */
+
+slash_command_group(env, "CSH environment variables");
+
+static int slash_var_set(struct slash *slash)
+{
+    optparse_t * parser = optparse_new("var_set", "NAME VALUE");
+    optparse_add_help(parser);
+    int argi = optparse_parse(parser, slash->argc - 1, (const char **) slash->argv + 1);
+    if (argi == -1) {
+        optparse_del(parser);
+        return SLASH_EUSAGE;
+    }
+    if ((slash->argc - argi) != 3) {
+        printf("Must give NAME and VALUE parameters\n");
+        optparse_del(parser);
+	    return SLASH_EINVAL;
+    }
+    csh_putvar(slash->argv[1], slash->argv[2]);
+    printf("Variable %s defined, value=%s\n", slash->argv[1], csh_getvar(slash->argv[1]));
+    optparse_del(parser);    
+	return SLASH_SUCCESS;
+}
+slash_command(var_set, slash_var_set, NULL, "Create or update an environment variable in CSH");
```

Comment:
I believe we would be setting a new precedence by using underscores in slash commands.
**slash_command(var_set, ...);** could alternatively be replaced with **slash_command_sub(var, set, ...);**
I am impartial to either pattern myself, but I wonder if I'm missing some reasoning here.

Reply:
`param_server` seems to be the counter example here...

Reply:
I can't argue with that ...